include/minstrobe.hpp

```diff
@@ -0,0 +1,531 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2021, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2021, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Hossein Eizadi Moghadam <hosseinem AT fu-berlin.de>
+ * \brief Provides minstrobe.
+ */
+
+#pragma once
+
+#include <seqan3/std/algorithm>
+#include <deque>
+#include <tuple>
+
+#include <seqan3/core/detail/empty_type.hpp>
+#include <seqan3/core/range/detail/adaptor_from_functor.hpp>
+#include <seqan3/core/range/type_traits.hpp>
+#include <seqan3/utility/range/concept.hpp>
+#include <seqan3/utility/type_traits/lazy_conditional.hpp>
+
+namespace seqan3::detail
+{
+// ---------------------------------------------------------------------------------------------------------------------
+// minstrobe_view class
+// ---------------------------------------------------------------------------------------------------------------------
+
+/*!\brief The type returned by minstrobe.
+ * \tparam urng1_t The type of the underlying range, must model std::ranges::forward_range, the reference type must
+ *                 model std::totally_ordered. The typical use case is that the reference type is the result of
+ *                 seqan3::kmer_hash.
+ * \tparam measure_distance If true, then not the actual minstrobes are returned, but the distances of the minstrobes.
+ * \implements std::ranges::view
+ * \ingroup search_views
+ *
+ *
+ * \note Most members of this class are generated by std::ranges::view_interface which is not yet documented here.
+
+ */
+template <std::ranges::view urng1_t>
+class minstrobe_view : public std::ranges::view_interface<minstrobe_view<urng1_t>>
+{
+private:
+    static_assert(std::ranges::forward_range<urng1_t>, "The minstrobe_view only works on forward_ranges.");
+    static_assert(std::totally_ordered<std::ranges::range_reference_t<urng1_t>>,
+                  "The reference type of the underlying range must model std::totally_ordered.");
+
+    //!\brief Whether the given ranges are const_iterable
+    static constexpr bool const_iterable = seqan3::const_iterable_range<urng1_t>;
+
+    //!\brief The first underlying range.
+    urng1_t urange1{};
+
+    //!\brief minimum, maximum, and order (n) values
+    size_t window_min{};
+    size_t window_max{};
+    size_t n{};
```

Comment:
Can you name this variable order_n instead of n?

Reply:
Since in this implementation order is presumed, I'll just remove it and call it minstrobe2.hpp?

Reply:
Sounds good. :)